src/snowflake/snowpark/modin/pandas/indexing.py

```diff
@@ -833,6 +836,11 @@ def _try_partial_string_indexing_for_string(
                 slice or the original `row_loc`
             """
             # TODO: SNOW-1063352: Modin upgrade - modin.pandas.indexing._LocIndexer
+            # Check if self or its index is a TimedeltaIndex. `index_dtypes` retrieves the dtypes of the index columns.
+            if is_timedelta64_dtype(self.df._query_compiler.index_dtypes[0]):
```

Comment:
I can switch it to `self.df.index.dtype` for now; I did not do that since I wasn't sure if the index would be pulled in locally because of `self.df.index`

Reply:
this increases the query counts - I think I will keep it as `self.df._query_compiler.index_dtypes[0]` to avoid that

Reply:
why it increased the query count? Weird to me.